segmentstore/server/src/main/java/io/pravega/segmentstore/server/DirectSegmentAccess.java

```diff
@@ -0,0 +1,134 @@
+     * @param cache        If set, then any Extended Attribute values that are not already in the in-memory Segment
+     *                     Metadata cache will be atomically added using a conditional update (comparing against a missing value).
+     *                     This argument will be ignored if the Segment is currently Sealed.
+     * @param timeout      Timeout for the operation.
+     * @return A Completable future that, when completed, will contain a Map of Attribute Ids to their latest values. Any
+     * Attribute that is not set will also be returned (with a value equal to Attributes.NULL_ATTRIBUTE_VALUE). If the operation
+     * failed, the future will be failed with the causing exception.
+     * @throws NullPointerException     If any of the arguments are null.
+     * @throws IllegalArgumentException If the Segment Name is invalid (NOTE: this doesn't check if the Segment
+     *                                  does not exist - that exception will be set in the returned CompletableFuture).
+     */
+    CompletableFuture<Map<UUID, Long>> getAttributes(Collection<UUID> attributeIds, boolean cache, Duration timeout);
+
+    /**
+     * Initiates a Read operation on the Segment and returns a ReadResult which can be used to consume the read data.
+     * @see StreamSegmentStore#read(String, long, int, Duration).
+     *
+     * @param offset    The offset within the Segment to start reading at.
+     * @param maxLength The maximum number of bytes to read.
+     * @param timeout   Timeout for the operation.
+     * @return A ReadResult instance that can be used to consume the read data.
+     * @throws NullPointerException     If any of the arguments are null.
+     * @throws IllegalArgumentException If any of the arguments are invalid.
+     */
+    ReadResult read(long offset, int maxLength, Duration timeout);
+
+    /**
+     * Gets information about the Segment.
+     * @see StreamSegmentStore#getStreamSegmentInfo(String, boolean, Duration).
+     *
+     * @return The requested Segment Info. Note that this result will only contain those attributes that
+     * are loaded in memory (if any) or Core Attributes. To ensure that Extended Attributes are also included, you must use
+     * getAttributes(), which will fetch all attributes, regardless of where they are currently located.
+     * @throws IllegalArgumentException If any of the arguments are invalid.
+     */
+    SegmentProperties getInfo();
+
+    /**
+     * Seals the Segment.
+     * @see StreamSegmentStore#sealStreamSegment(String, Duration).
+     *
+     * @param timeout Timeout for the operation
+     * @return A CompletableFuture that, when completed normally, will contain the final length of the Segment.
+     * If the operation failed, the future will be failed with the causing exception.
+     * @throws IllegalArgumentException If any of the arguments are invalid.
+     */
+    CompletableFuture<Long> seal(Duration timeout);
```

Comment:
Why are you adding `seal` and `truncate`, but not `delete` segment? Deleting the segment would invalidate a direct access object, but that would happen anyways in the case the segment is deleted concurrently. 

In any case, please share your thoughts.

Reply:
I don't need `delete` here. If I do, I will add it. I need all the other operations for the work that is about to happen.